x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -502,6 +503,19 @@ private void authorizeAction(
                                 indicesAndAliasesResolver.resolve(action, request, projectMetadata, authorizedIndices)
                             ),
                             e -> {
+                                if (e instanceof InvalidIndexNameException) {
```

Comment:
Another exception I've considered was `InvalidArgumentException` but decided to skip it since it's not guaranteed that it's caused by the user's invalid input (sometimes it hides internal errors).

Reply:
Yeah, `InvalidArgumentException` feels too broad to catch here... 

WDYT about adding `InvalidSelectorException` (extend `InvalidArgumentException`) and throw that in `getByKeyOrThrow` and also throw either `InvalidSelectorException` or `InvalidIndexNameException` inside `ensureNoSelectorsProvided`? I think that would cover a decent amount of other cases.